src/D2L.Security.OAuth2/D2L.Security.OAuth2.csproj

```diff
@@ -16,18 +16,37 @@
 		<PackageReference Include="D2L.CodeStyle.Analyzers">
 			<PrivateAssets>All</PrivateAssets>
 		</PackageReference>
+
 		<PackageReference Include="D2L.CodeStyle.Annotations" />
 		<PackageReference Include="D2L.Services.Core.Exceptions" />
-		<PackageReference Include="Newtonsoft.Json" />
+
+		<!-- In .NET Framework use Newtonsoft.Json.
+		     Otherwise, use System.Text.Json. -->
+		<PackageReference Include="Newtonsoft.Json" Condition="'$(TargetFramework)' != 'net50'" />
+
+		<!-- In .NET Framework use System.Runtime.Caching.MemoryCache.
+		     Otherwise, use the NuGet package. -->
+		<Reference Include="System.Runtime.Caching" Condition="'$(TargetFramework)' != 'net50'" />
+		<PackageReference Include="Microsoft.Extensions.Caching.Memory" Condition="'$(TargetFramework)' == 'net50'" />
+
 		<PackageReference Include="System.Collections.Immutable" />
 		<PackageReference Include="System.IdentityModel.Tokens.Jwt" />
 	</ItemGroup>
 
 	<ItemGroup>
 		<Reference Include="System.IdentityModel" />
 		<Reference Include="System.Net.Http" />
-		<Reference Include="System.Runtime.Caching" />
 		<Reference Include="System.Web" />
 	</ItemGroup>
 
+	<ItemGroup Condition="'$(TargetFramework)' == 'net50'">
+		<Compile Remove="**/*.net4x.cs" />
+		<None Include="**/*.net4x.cs" />
+	</ItemGroup>
+
+	<ItemGroup Condition="'$(TargetFramework)' != 'net50'">
+		<Compile Remove="**/*.net50.cs" />
+		<None Include="**/*.net50.cs" />
+	</ItemGroup>

```

Comment:
This is an alternative to `#if`s in each file.

This works really nice in VS: F5 builds all targets, auto-complete works well in every file (as if the right assemblies are referenced for the right files). The `<None Include=` are there so they show up in the solution browser, which is only rendering for the `net472` build. Somehow VS and intellisense when you're actually using the `.net50.cs` files works they way you'd wish it to though.